### ISSUE TITLE
Depth of the currently traversed element in constant time

### ISSUE THREAD

**Issue by mgoldenbe** [joaonuno/tree-model-js#94]
Can I access the depth of the currently traversed (e.g. during the depth-first walk) element **in constant time**? Right now, the only way of getting the depth that I see (besides storing it with the node) is by calling `getPath`:

`root.walk(el => {console.log(el.model.id, el.getPath().length);})`

**Comment by joaonuno** [comment]
You're right, that's the only way as of now and it's not constant time as you noted. I'll leave this open because it seems a nice improvement for a future version.

**Comment by coreymunn3** [comment]
Wow, if it weren't for this issue I wouldn't have even know that getting depth was possible using this library, or that you even get something with a "getPath" method on it. The documentation here is painfully inadequate.

**Comment by joaonuno** [comment]
Hi, `getPath` is documented [here](https://github.com/joaonuno/tree-model-js#get-the-node-path) and there's an example on the lib's [page](http://jnuno.com/tree-model-js/):
![image](https://user-images.githubusercontent.com/1133687/158651130-36f1af73-44b4-4a94-a01e-92370dfa8d21.png)
